handlers.js

```diff
@@ -191,17 +185,11 @@ async function handle_message(reply, content) {
     handler = handlers[content.split(" ")[0]];
     if (handler != undefined) {
         try{
-            let r;
-            if (handler.name == reply_wait().name) {
-                r = await handler(reply, content.split(" ").slice(1, 9));
-            }
-            else {
-                r = await handler(content.split(" ").slice(1, 9));
-            }
+            let r = await handler(content.split(" ").slice(1, 9), reply);
             if (!r) {
                 reply(format.format_error("Invalid arguments"));
             } else {
-                reply(r);
+                reply(...typeof r[Symbol.iterator] == "function" ? r : [r]);
```

Comment:
Not a huge fan of this check, not sure how else to handle `reply_wait` returning multiple values.

Reply:
Yeah it's definitely a bit strange but something like this is needed I guess. Can you just put in a comment to explain why you're doing this in case someone tries to come back and understand? It's a bit hard to parse without proper context.